velox/common/memory/MemoryPool.cpp

```diff
@@ -805,12 +803,8 @@ bool MemoryPoolImpl::incrementReservationThreadSafe(
       treeMemoryUsage()));
 }
 
-bool MemoryPoolImpl::maybeIncrementReservation(uint64_t size) {
-  std::lock_guard<std::mutex> l(mutex_);
-  return maybeIncrementReservationLocked(size);
-}
-
 bool MemoryPoolImpl::maybeIncrementReservationLocked(uint64_t size) {
```

Comment:
We shall remove maybeIncrementReservationLocked but keep maybeIncrementReservation. maybeIncrementReservationLocked means the function is called with lock.